src/lib/messenger/fullwidth.ts

```diff
@@ -0,0 +1,44 @@
+import { isObject } from '@guardian/libs';
+import fastdom from 'fastdom';
+import type { RegisterListener } from 'core/messenger';
+
+interface FullwidthSpecs {
+	fullwidth: boolean;
+}
+
+const isFullwidthSpecs = (specs: unknown): specs is FullwidthSpecs =>
+	!!specs && isObject(specs) && 'fullwidth' in specs;
+
+const fullwidth = (specs: FullwidthSpecs, slot: HTMLElement) =>
+	fastdom.mutate(() => {
+		if (specs.fullwidth) {
+			slot.classList.add('ad-slot--full-width');
+		} else {
+			slot.classList.remove('ad-slot--full-width');
+		}
+	});
+
+const init = (register: RegisterListener): void => {
+	register('fullwidth', (specs, ret, iframe) => {
+		if (iframe && specs) {
+			if (!isFullwidthSpecs(specs)) {
+				return;
+			}
+			const adSlot =
+				iframe.closest<HTMLElement>('.js-ad-slot') ?? undefined;
+
+			const name = adSlot?.dataset.name;
+
+			// only allow for banner ads
+			if (!name?.startsWith('fronts-banner') || !adSlot) {
```

Comment:
if `adslot` is falsy then `name` will be undefined, so I don't think you need this second check

Reply:
I agree it's a bit pointless but typescript complains without it when calling `fullwidth`